packages/api/src/reefs/reefs.service.ts

```
@@ -280,6 +274,90 @@ export class ReefsService {
     };
   }
 
+  async deploySpotter(id: number, deploySpotterDto: DeploySpotterDto) {
+    const { endDate } = deploySpotterDto;
+
+    const reef = await this.reefsRepository.findOne({
+      id,
+      spotterId: Not(IsNull()),
+      status: Not(ReefStatus.Deployed),
+    });
+
+    if (!reef) {
+      throw new NotFoundException(`Reef with ID ${id} not found`);
+    }
```

This is super confusing. Let's only get the reef by ID before throwing `throw new NotFoundException(`Reef with ID ${id} not found`);` if we can't find it.

THEN check of there is actually a spotter and the status is not already deployed and set a clearer message.

---

Added two separate error message:
1. Reef is already deployed
2. Reef has no spotter